Compatibility issue (minor), error message in case of invalid repo.

Compatibility issue (minor), error message in case of invalid repo.

Old version

	jk@paju:~$ borg  --version
	borg 1.1.8

Repo dir is not valid repo:

	jk@paju:~$ borg create ~/tmp::foo ~/tmp
	/home/jk/tmp is not a valid repository. Check repo config.


Repo dir does not exist:

	jk@paju:~$ borg create ~/tmp/foo::foo ~/tmp
	Repository /home/jk/tmp/foo does not exist.

Latest version

	jk@tiira:~$ borg --version
	borg 1.1.9

Repo dir is not valid repo:

	jk@tiira:~$ borg create ~/tmp::foo ~/tmp
	Exception ignored in: <bound method Repository.__del__ of <Repository /home/jk/tmp>>
	Traceback (most recent call last):
	  File "/usr/lib/python3/dist-packages/borg/repository.py", line 179, in __del__
	    assert False, "cleanup happened in Repository.__del__"
	AssertionError: cleanup happened in Repository.__del__
	Local Exception
	Traceback (most recent call last):
	  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 4455, in main
	    exit_code = archiver.run(args)
	  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 4387, in run
    	return set_ec(func(args))
	  File "/usr/lib/python3/dist-packages/borg/archiver.py", line 139, in wrapper
    	with repository:
	  File "/usr/lib/python3/dist-packages/borg/repository.py", line 189, in __enter__
	    self.open(self.path, bool(self.exclusive), lock_wait=self.lock_wait, lock=self.do_lock)
	  File "/usr/lib/python3/dist-packages/borg/repository.py", line 396, in open
	    with open(os.path.join(self.path, 'config')) as fd:
	FileNotFoundError: [Errno 2] No such file or directory: '/home/jk/tmp/config'
	
	Platform: Linux tiira 4.4.0-142-generic #168-Ubuntu SMP Wed Jan 16 21:00:45 UTC 2019 x86_64
	Linux: Ubuntu 16.04 xenial
	Borg: 1.1.9  Python: CPython 3.5.2
	PID: 23536  CWD: /home/jk
	sys.argv: ['/usr/bin/borg', 'create', '/home/jk/tmp::foo', '/home/jk/tmp']
	SSH_ORIGINAL_COMMAND: None

Repo dir does not exist:
	
	jk@tiira:~$ borg create ~/tmp/foo::foo ~/tmp
	Repository /home/jk/tmp/foo does not exist.


Please restore the v1.1.8 type of error message in case of invalid repo dir.

--- a/src/borg/repository.py
+++ b/src/borg/repository.py
@@ -393,8 +393,12 @@ class Repository:
         else:
             self.lock = None
         self.config = ConfigParser(interpolation=None)
-        with open(os.path.join(self.path, 'config')) as fd:
-            self.config.read_file(fd)
+        try:
+            with open(os.path.join(self.path, 'config')) as fd:
+                self.config.read_file(fd)
+        except FileNotFoundError:
+            self.close()
+            raise self.InvalidRepository(self.path)
         if 'repository' not in self.config.sections() or self.config.getint('repository', 'version') != 1:
             self.close()
             raise self.InvalidRepository(path)

